### DISCUSSION TITLE
npx prisma introspect --force isn't working on docker

### DISCUSSION THREAD

**Discussion by ArunKadariCS** [prisma/prisma#7812] [Q&A] [answered]
On the local machine the command `npx prisma introspect--force` is working as expected but on the docker it's throwing the below error
```
Step 8/10 : RUN npx prisma introspect --force
 ---> Running in ff2f1297b22c
Prisma schema loaded from prisma/schema.prisma

Introspecting based on datasource defined in prisma/schema.prisma …

Error: P1012 Introspection failed as your current Prisma schema file is invalid

Please fix your current schema manually, use prisma validate to confirm it is valid and then run this command again.
Or run this command with the --force flag to ignore your current schema and overwrite it. All local modifications will be lost.

npm notice 
npm notice New minor version of npm available! 7.15.1 -> 7.18.1
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.18.1>
npm notice Run `npm install -g npm@7.18.1` to update!
npm notice 
The command '/bin/sh -c npx prisma introspect --force' returned a non-zero code: 1

Error: Process completed with exit code 1.

```

**Comment by ryands17** [comment]
@ArunKadariCS 👋 

Could you share what steps did you take to create this? Especially the `Dockerfile`?

**Reply by ArunKadariCS** [reply]
Here is the github actions workflow yaml file configuration. 
```yml
env:
  SERVICE_NAME: ${{ github.event.repository.name }}
  RUN_REGION: us-west1-c
  COMMIT_SHA: ${{ github.sha }}
  PORT: 8080
  DEV_DATABASE_URL: ***********************************************
```

.............................  And  in the `Dockerfile` the `DEV_DATABASE_URL` is commented out. ..............................

```Dockerfile
FROM node:latest as build
	# set the working direction
	WORKDIR /app
	

	# add `/app/node_modules/.bin` to $PATH
	ENV PATH /app/node_modules/.bin:$PATH
	#ENV DEV_DATABASE_URL: ***********************************************************

	# install app dependencies
	COPY package.json ./
	COPY package-lock.json ./
	COPY . ./
	RUN npm install
	RUN npx prisma introspect --force
	RUN npx prisma generate
	CMD "npm" "start"
	#RUN prisma validate

```
Also I'm already running the introspect command on the local machine before pushing the code to deployment.  Should  the deployment  also run the introspect command. My understanding is that introspect command will generate the schema and we don't need to run the introspect again on the deployment.

**Reply by ryands17** [reply, marked as answer]
If you're already running introspect in the local machine, it is not required to run it in the deployment. Introspect is a one-time operation as that just maps `schema.prisma` to the latest database structure.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
